store.go

```diff
@@ -1730,11 +1819,26 @@ func (s *store) DeleteImage(id string, commit bool) (layers []string, err error)
 					}
 				}
 			}
+			if layer == image.TopLayer {
+				for _, mappedLayer := range image.MappedTopLayers {
+					if childList, ok := childrenByParent[mappedLayer]; ok && childList != nil {
+						children := *childList
+						for _, child := range children {
+							if child != lastRemoved {
+								otherRefs++
```

Comment:
Seems like this should be a function and just return at this point.

Reply:
Made it a function.